client/src/pages/StudentView.js

```diff
@@ -18,6 +18,114 @@ const StudentView = () => {
+    const [updatedNextSaturday, setUpdatedNextSaturday ] = useState(0);//If next(ending day which is newer) has changed by user this useState keeps track of it
+	const [isPrevClicked,setIsPrevClicked] = useState(false);
+	const [mountedPrev, setMountedPrev] = useState(false);
+	const [isNextClicked,setIsNextClicked] = useState(false);
+	const [mountedNext, setMountedNext] = useState(false);
+
+
+	// Calculating the timestamps for current Sat to Sat period
+	function getTimestampsOfPreviousAndNextSaturday() {
+		const now = new Date(); // Get the current date and time
+		const dayOfWeek = now.getUTCDay(); // Get the day of the week (0 = Sunday, 1 = Monday, ..., 6 = Saturday)
+		const saturday = new Date(now); // Create a new Date object based on the current date and time
+		saturday.setUTCDate(saturday.getUTCDate() - dayOfWeek + 6); // Set the date to the next Saturday
+		const nextSaturday = new Date(saturday); // Create a new Date object based on the next Saturday
+		const previousSaturday = new Date(saturday); // Create a new Date object based on the next Saturday
+		previousSaturday.setUTCDate(previousSaturday.getUTCDate() - 7); // Set the date to the previous Saturday
+		const timestampOfPreviousSaturday = (previousSaturday.getTime())/1000; // Get the timestamp of the previous Saturday in milliseconds since epoch
+		const timestampOfNextSaturday = (nextSaturday.getTime())/1000; // Get the timestamp of the next Saturday in milliseconds since epoch
+		return [timestampOfPreviousSaturday, timestampOfNextSaturday];
+	}
+    
+	// Calculating timestamp for 7 days ago
+    function getTimestampOfSevenDaysAgo(timeStamp) {
+			const now = timeStamp; // Get the current timestamp in milliseconds
+			const sevenDaysInMs = 7 * 24 * 60 * 60 ; // Calculate 7 days in milliseconds
+			const timestampOfSevenDaysAgo = now - sevenDaysInMs; // Subtract 7 days from the current timestamp
+			return timestampOfSevenDaysAgo;
+	}
+
+	// Calculating timestamp for 7 later
+    function getTimestampOfSevenDaysAfter(timeStamp) {
+		const now = timeStamp; // Get the current timestamp in milliseconds
+		const sevenDaysInMs = 7 * 24 * 60 * 60 ; // Calculate 7 days in milliseconds
+		const timestampOfSevenDaysAgo = now + sevenDaysInMs; // Subtract 7 days from the current timestamp
+		return timestampOfSevenDaysAgo;
+	}
+
+	// Getting data when first page is loaded
+    useEffect( ()=>{
+		// Find current Sat to Sat timestamps where prev is the older Sat
+      	const currentWeekStamps = getTimestampsOfPreviousAndNextSaturday();	
+		setUpdatedPreviousSaturday(currentWeekStamps[0]);
+		setUpdatedNextSaturday(currentWeekStamps[1]);
+        
+		// When it loads for the first time it fetch data for general, we can change it however we want.
+		// let URL = `/api/data/general?oldest=${currentWeekStamps[0]}&latest=${currentWeekStamps[1]}`;
```

Comment:
Hey can you remove the unnecessary comments? From line 67-78?

Reply:
Done